src/eei.cpp

```diff
@@ -48,15 +48,85 @@ Literal EthereumInterface::callImport(Import *import, LiteralList& arguments) {
     if (import->base == Name("getAddress")) {
       cout << "getAddress ";
 
-      uint32_t resultOffset = arguments[0].geti32();
+      const uint32_t resultOffset = arguments[0].geti32();
 
       cout << resultOffset << "\n";
 
-      copyAddressToMemory(msg.address, resultOffset);
+      storeUint160(resultOffset, msg.address);
 
       return Literal();
     }
 
+    if (import->base == Name("getBalance")) {
+      std::cout << "getbalance\n";
+
+      uint32_t addressOffset = arguments[0].geti32();
+      uint32_t resultOffset = arguments[1].geti32();
+
+      evm_uint256be result;
+      evm_address address = loadUint160(addressOffset);
+
+      context.fn_table->get_balance(&result, &context, &address);
+      storeUint128(resultOffset, result);
+
+      return Literal();
+    }
+
+    if (import->base == Name("getBlockHash")) {
+      std::cout << "getblockhash ";
+
+      int64_t number = arguments[0].geti64();
+      uint32_t resultOffset = arguments[1].geti32();
+
+      std::cout << number << "\n";
+
+      evm_uint256be blockhash;
+      context.fn_table->get_block_hash(&blockhash, &context, number);
+      storeUint256(resultOffset, blockhash);
+
+      return Literal();
+    }
+
+    if (import->base == Name("call")) {
+      std::cout << "call\n";
+
+      int64_t gas = arguments[0].geti64();
+      uint32_t addressOffset = arguments[1].geti32();
+      uint32_t valueOffset = arguments[2].geti32();
+      uint32_t dataOffset = arguments[3].geti32();
+      int32_t dataLength = arguments[4].geti32();
+      uint32_t resultOffset = arguments[5].geti32();
+      int32_t resultLength = arguments[6].geti32();
+
+      evm_result *call_result;
+      evm_message call_message;
+      int32_t vm_status = 0;
+      uint8_t *input_data = new uint8_t[dataLength];
```

Comment:
You should not use `new` / `delete` directly. Here you can use `std::vector input_data(dataLength)`.

But it might be better to actually `loadMemory` to return `std::vector` with the memory copy.